--- karma/include/os.h.orig	2001-11-14 03:01:56.000000000 +0100
+++ karma/include/os.h	2004-01-27 20:24:35.121888953 +0100
@@ -630,6 +630,21 @@
 #define IS_ALIGNED(ptr,size) ((uaddr) ptr % size == 0)
 #endif
 
+/* AMD64 */
+#ifdef MACHINE_amd64
+#define MACHINE_SUPPORTED
+#define MACHINE_LITTLE_ENDIAN
+#define HAS_IEEE
+#define CAN_ZERO_CLEAR
+#define Kword16s short
+#define Kword16u unsigned short
+#define Kword32s int
+#define Kword32u unsigned int
+#define Kword64s long
+#define Kword64u unsigned long
+#define IS_ALIGNED(ptr,size) ((uaddr) ptr % size == 0)
+#endif
+
 /*  Intel i860  */
 #ifdef MACHINE_i860
 #define MACHINE_SUPPORTED
